07_safety/reported-crime/reported-crime-place-all.qmd

```diff
@@ -0,0 +1,557 @@
+  )
+
+```
+
+A very small percentage of crime is reported by special agencies (agency_type == 5) 
+```{r}
+offenses_agency_place %>%
+  group_by(agency_type) %>%
+  summarize(
+    n = n(),
+    n_rpt = sum(!is.na(all)),
+    crime = sum(all, na.rm = TRUE)
+  ) %>%
+  ungroup() %>%
+  mutate(
+    pct = crime / sum(crime),
+    pct_report = n_rpt / n
+  )
+  
+```
+
+### 4.3 Summarize by place
+Count number of crimes and number of reporting agencies at the place-level.
+
+Note that this section calculates the percent of agencies reporting violent crimes and property crimes separately from all crimes because the 2022 update requires separate quality measures for each rate. However, all agencies either report both types of crime or no crime, so the numbers are identical.
+```{r}
+offenses_place <- offenses_agency_place %>%
+  group_by(year, state, place) %>%
+  summarize(
+    n = n(),
+    n_reporting_all = sum(reporting_all),
+    n_reporting_violent = sum(reporting_violent),
+    n_reporting_property = sum(reporting_property),
+    n_core_city = sum(core_city),
+    n_core_city_rpt = sum(core_city==1 & reporting_all==1),
+    across(all:property, ~sum(.x, na.rm=TRUE))
+  ) %>%
+  mutate(
+    agencies_reporting_all = n_reporting_all / n,
+    agencies_reporting_violent = n_reporting_violent / n,
+    agencies_reporting_property = n_reporting_property / n,
+    core_reporting = n_core_city_rpt / n_core_city,
+    GEOID = str_c(state, place)
+  ) %>%
+  ungroup()
+
+```
+
+
+## 5. Calculate rates of reported property crimes per 100,000 people and reported violent crimes per 100,000 people
+This section calculates the violent crime rate and property crime rate. Rates are calculated as the number of crimes per 100,000 people using ACS place populations.
+
+Crime Rate = (Crime Count / Population) * 100,000
+
+### 5.1 Load demographic information
+First load the demographic information for each place and for each year 2016-2022. This file was created using 5 year ACS data in `01_agency_geo_place.Rmd`.
+```{r}
+place_demo <- read_csv(here::here("07_safety", "modified data", "all_place_demo.csv"))
+
+```
+
+### 5.2 Join place-level offense data onto place-level population data
+```{r}
+offense_place_demo <- left_join(
+  x = place_demo %>%
+    select(c(year, GEOID, total_people)),
+  y = offenses_place,
+  by = c("year", "GEOID")
+)
+  
+```
+
+
+### 5.3 Calculate crime rates
+Calculate place-level rates of reported violent crimes per 100,000 people and reported property crimes per 100,000 people.
+```{r}
+rates_place <- offense_place_demo %>%
+  mutate(
+    rate_crime_violent = violent / total_people * 100000,
+    rate_crime_property = property / total_people * 100000
+  )
+
+```
+
+Suppress rates for places that have population less than 30 people.
+Note that no places have a population of less than 30. Future updates may consider changing the population threshold for suppression.
+```{r}
+rates_place <- rates_place %>%
```

Comment:
`rates_place <- rates_place %>%
  mutate(
    rate_crime_violent = ifelse(total_people < 30 | n_reporting_all == 0, NA, rate_crime_violent),
    rate_crime_property = ifelse(total_people < 30| n_reporting_all == 0, NA, rate_crime_property)
  )`

Reply:
Thank you!! I ended up including the `ifelse` condition directly in the code that creates the rates, but got the same effect.